sources/system/file-system/file-system.dylan

```diff
@@ -516,3 +516,19 @@ define sideways method list-locator
      locator);
   locators
 end method list-locator;
+
+///
+
+define generic symbolic-link
+  (target :: <pathname>, link :: <pathname>) => ();
+
+define method symbolic-link
+    (target :: <file-system-locator>, link :: <file-system-locator>) => ()
+  %symbolic-link(target, link)
+end method symbolic-link;
+
+define method symbolic-link 
+    (target :: <string>, link :: <string>) => ()
```

Comment:
Specialize on `<pathname>, <pathname>` here to cover all the dispatch cases.

Reply:
I don't understand it. `<pathname>` is either a `<string>` or a `<file-system-locator>`. I have specialize the gf in both types.

Reply:
You are covering `string X string` and `locator X locator` but there is also `string X locator` and `locator X string`.  

On second thought you can actually do it all with one method on `pathname X pathname` as long as you call `as(<file-system-locator>, arg)` on each method argument.